File: parse_manuals.js
const fs = require('fs');
const pdf = require('pdf-parse');
const path = require('path');

const pdfDir = path.join(__dirname, '..');
const outDir = path.join(__dirname, 'manuals');

if (!fs.existsSync(outDir)) {
    fs.mkdirSync(outDir);
}

const filesToParse = [
    'Case Brief Template.pdf',
    'Interview Guide.pdf',
    'Scoring Form.pdf',
    'Scoring Manual.pdf'
];

async function parseAll() {
    for (const file of filesToParse) {
        const filePath = path.join(pdfDir, file);
        if (fs.existsSync(filePath)) {
            const dataBuffer = fs.readFileSync(filePath);
            const data = await pdf(dataBuffer);
            const textPath = path.join(outDir, file.replace('.pdf', '.txt'));
            fs.writeFileSync(textPath, data.text);
            console.log(`Parsed ${file} -> ${data.text.length} characters`);
        } else {
            console.error(`Missing file: ${filePath}`);
        }
    }
}

parseAll().catch(console.error);

File: parse_manuals.py
import os
from pypdf import PdfReader

pdf_dir = '../'
out_dir = './manuals'
os.makedirs(out_dir, exist_ok=True)

files = [
    'Case Brief Template.pdf',
    'Interview Guide.pdf',
    'Scoring Form.pdf',
    'Scoring Manual.pdf'
]

for file in files:
    path = os.path.join(pdf_dir, file)
    if os.path.exists(path):
        reader = PdfReader(path)
        text = ''
        for page in reader.pages:
            text += page.extract_text() + '\n'
        with open(os.path.join(out_dir, file.replace('.pdf', '.txt')), 'w', encoding='utf-8') as f:
            f.write(text)
        print(f"Parsed {file}")
    else:
        print(f"Missing {file}")
